### PR TITLE
fix: skip-audit-log-for-soft-deleted-CR

### LINKED ISSUE #6248: Fix missing `live_from` when creating audit log
Sentry Issue: [FLAGSMITH-API-5D4](https://flagsmith.sentry.io/issues/7000304068/?referrer=github_integration)

```
TypeError: '<' not supported between instances of 'datetime.datetime' and 'NoneType'
(6 additional frame(s) were not displayed)
...
  File "features/workflows/core/models.py", line 198, in create_audit_log_for_related_feature_state
    if self.committed_at < feature_state.live_from:  # type: ignore[operator]

TypeError: '<' not supported between instances of 'datetime.datetime' and 'NoneType'
(20 additional frame(s) were not displayed)
...
  File "features/workflows/core/models.py", line 198, in create_audit_log_for_related_feature_state
    if self.committed_at < feature_state.live_from:  # type: ignore[operator]
```

### PR DESCRIPTION
**Pull request by Zaimwa9**
Thanks for submitting a PR! Please check the boxes below:

- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

Closes #6248 

## Context

Edge-case that is a combination between data corruption when migrating to `feature versioning` and trying to delete a project.
When migrating, feature states from committed CR where linked to EFV with:
- A change_request_id
- `live_from = None`
When deleting the project, the soft-delete was triggering the hook [here](https://github.com/Flagsmith/flagsmith/blob/main/api/features/workflows/core/models.py#L195-L206) without `live_from` which raised the NoneType error

## Changes
- Updated hook `AFTER_SAVE` to ignore CR with `deleted_at`

## How did you test this code?
- Added a test (fails without change)
- Manually
1. Create a new project (feature_versioning_v2 disabled)
2. Create and publish a change request now (it must have existing CR)
3. Create a change request in the future
4. Publish it
5. Enable feature_versioning_v2
6. Delete the project

### CODE REVIEW

File: api/features/versioning/tasks.py [hunk 1]
```diff
@@ -106,7 +106,10 @@ def _create_initial_feature_versions(environment: "Environment"):  # type: ignor
             feature_states__in=latest_feature_states
         )
 
-        latest_feature_states.update(environment_feature_version=ef_version)
+        latest_feature_states.update(
+            environment_feature_version=ef_version,
+            change_request=None,
```
Comment (Zaimwa9) [inline api/features/versioning/tasks.py]: I believe this is the less risky place to fix this issue.

I preferred not to add a guard against `live_from = None` because for any similar bug it would have entered the wrong branch and create an irrelevant log `create_feature_state_updated_by_change_request_audit_log`

**Review by khvn26 — APPROVED** [review]
LGTM
